authorization/permission/model/permission_model_service_blackbox_test.go

```diff
@@ -0,0 +1,687 @@
+	require.NoError(s.T(), err, "Could not create resource")
+	if err != nil {
+		return nil, err
+	}
+
+	createdResource, err := s.resourceRepo.Load(s.Ctx, resource.ResourceID)
+	require.NoError(s.T(), err, "Could not load resource")
+	if err != nil {
+		return nil, err
+	}
+
+	return createdResource, nil
+}
+
+// Assigns a role to a user for a resource
+func (s *permissionModelServiceBlackBoxTest) assignRoleForResource(resource resource.Resource, identity account.Identity, role roleRepo.Role) error {
+	err := s.identityRoleRepo.Create(s.Ctx, &identityrole.IdentityRole{
+		IdentityID: identity.ID,
+		ResourceID: resource.ResourceID,
+		RoleID:     role.RoleID,
+	})
+	require.NoError(s.T(), err, "Could not assign role to identity")
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Create a test resource of the default type and assign the default role to the specified user
+func (s *permissionModelServiceBlackBoxTest) createTestResourceAndAssignDefaultRole(identity account.Identity) (*resource.Resource, error) {
+	resource, err := s.createTestResource(testResourceTypeArea)
+	require.NoError(s.T(), err, "Could not create resource")
+
+	err = s.assignRoleForResource(*resource, identity, s.testAreaRole)
+	require.NoError(s.T(), err, "Could not assign role to resource")
+
+	return resource, nil
+}
+
+// Create a test resource of the default type with no assigned permissions
+func (s *permissionModelServiceBlackBoxTest) createTestResourceWithNoPermissions(identity account.Identity) (*resource.Resource, error) {
+	// Lookup our default resource type
+	resourceType, err := s.resourceTypeRepo.Lookup(s.Ctx, testResourceTypeArea)
+	require.NoError(s.T(), err, "Could not lookup resource type")
+
+	resource := &resource.Resource{
+		ResourceID:       uuid.NewV4().String(),
+		ParentResourceID: nil,
+		ResourceType:     *resourceType,
+		ResourceTypeID:   resourceType.ResourceTypeID,
+	}
+
+	err = s.resourceRepo.Create(s.Ctx, resource)
+	require.NoError(s.T(), err, "Could not create resource")
+	if err != nil {
+		return nil, err
+	}
+
+	createdResource, err := s.resourceRepo.Load(s.Ctx, resource.ResourceID)
+	require.NoError(s.T(), err, "Could not load resource")
+	if err != nil {
+		return nil, err
+	}
+
+	return createdResource, nil
+}
+
+// Creates a child resource of the specified parent resource, of the specified type
+func (s *permissionModelServiceBlackBoxTest) createTestChildResource(parentResource resource.Resource, resourceTypeName string) (*resource.Resource, error) {
+	// Lookup the resource type
+	resourceType, err := s.resourceTypeRepo.Lookup(s.Ctx, resourceTypeName)
+	require.NoError(s.T(), err, "Could not lookup resource type")
+
+	resource := &resource.Resource{
+		ResourceID:       uuid.NewV4().String(),
+		ParentResourceID: &parentResource.ResourceID,
+		ResourceType:     *resourceType,
+		ResourceTypeID:   resourceType.ResourceTypeID,
+	}
+
+	err = s.resourceRepo.Create(s.Ctx, resource)
```

Comment:
`test/authorization.go` has `CreateTestResource(..)` 
might help you avoid initializing `resourceRepo` here.

Reply:
Fixed, thanks!